app/src/main/java/de/westnordost/streetcomplete/map/components/PinsMapComponent.kt

```diff
@@ -1,50 +1,23 @@
 package de.westnordost.streetcomplete.map.components
 
-import android.content.Context
-import android.graphics.PointF
-import android.graphics.drawable.BitmapDrawable
-import androidx.annotation.DrawableRes
 import com.mapzen.tangram.MapData
 import com.mapzen.tangram.geometry.Point
-import de.westnordost.streetcomplete.R
 import de.westnordost.streetcomplete.data.osm.mapdata.LatLon
-import de.westnordost.streetcomplete.ktx.getBitmapDrawable
-import de.westnordost.streetcomplete.ktx.toDp
 import de.westnordost.streetcomplete.map.tangram.KtMapController
-import de.westnordost.streetcomplete.map.tangram.Marker
 import de.westnordost.streetcomplete.map.tangram.toLngLat
 
-/** Takes care of displaying pins on the map and displaying pins as selected */
-class PinsMapComponent(private val ctx: Context, private val ctrl: KtMapController) {
+/** Takes care of displaying pins on the map */
+class PinsMapComponent(ctrl: KtMapController) {
 
-    private val pinsLayer: MapData
-    private val selectedPinsLayer: MapData
+    private val pinsLayer: MapData = ctrl.addDataLayer(PINS_LAYER)
 
-    private val pinSelectionMarkers: MutableList<Marker> = mutableListOf()
-
-    private val selectionDrawable: BitmapDrawable
-    private val selectionDrawableSize: PointF
-
-    init {
-        selectionDrawable = ctx.resources.getBitmapDrawable(R.drawable.pin_selection_ring)
-        selectionDrawableSize = PointF(
-            selectionDrawable.intrinsicWidth.toFloat().toDp(ctx),
-            selectionDrawable.intrinsicHeight.toFloat().toDp(ctx)
-        )
-
-        pinsLayer = ctrl.addDataLayer(PINS_LAYER)
-        selectedPinsLayer = ctrl.addDataLayer(SELECTED_PINS_LAYER)
-    }
-
-    fun clear() {
-        clearPins()
-        clearSelectedPins()
-    }
-
-    /* ------------------------------------------  Pins ----------------------------------------- */
+    /** Shows/hids the pins */
+    var isVisible: Boolean
+        get() = pinsLayer.visible
+        set(value) { pinsLayer.visible = value }
```

Comment:
Note: you could also do this (https://kotlinlang.org/docs/delegated-properties.html#delegating-to-another-property):

```suggestion
    var isVisible: Boolean by pinsLayer::visible
```
I'm not doing a full review, just a quick skim of a couple commits.

Reply:
Interesting, didn't know that. But it should be consistent among the codebase, so this should be done in a separate PR. there are more places other than that where it is done